### ISSUE TITLE
Persistent crash when launching a flame arrow into the Twilight Forest Snow Queen

### ISSUE THREAD

**Issue by Knifepony** [ACGaming/UniversalTweaks#137]
The Twilight Forest contains a boss called the Snow Queen whose ice shield acts much like a 360 shield. This likely means that projectiles are calculated similar to shields. Launching an arrow with the flame enchant or launching an arrow through lava into the flame queen (basically, a fire arrow) towards the flame queen and having it bounce off her shield will cause a crash.
Log attached below.
```
[16:33:21] [Server thread/ERROR] [minecraft/MinecraftServer]: Encountered an unexpected exception
net.minecraft.util.ReportedException: Ticking entity
    at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:773) ~[MinecraftServer.class:?]
    at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:668) ~[MinecraftServer.class:?]
    at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer.java:185) ~[chd.class:?]
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
    at java.lang.Thread.run(Unknown Source) [?:1.8.0_333]
Caused by: java.lang.ClassCastException: twilightforest.entity.boss.EntityTFSnowQueenIceShield cannot be cast to net.minecraft.entity.EntityLivingBase
    at net.minecraft.entity.projectile.EntityArrow.redirect$zcd000$utBlockFire(EntityArrow.java:731) ~[aeh.class:?]
    at net.minecraft.entity.projectile.EntityArrow.onHit(EntityArrow.java:366) ~[aeh.class:?]
    at net.minecraft.entity.projectile.EntityArrow.onUpdate(EntityArrow.java:267) ~[aeh.class:?]
    at net.minecraft.entity.projectile.EntityTippedArrow.onUpdate(SourceFile:98) ~[afa.class:?]
    at net.minecraft.world.World.updateEntityWithOptionalForce(World.java:1996) ~[amu.class:?]
    at net.minecraft.world.WorldServer.updateEntityWithOptionalForce(WorldServer.java:832) ~[oo.class:?]
    at net.minecraft.world.World.updateEntity(World.java:1958) ~[amu.class:?]
    at net.minecraft.world.World.updateEntities(World.java:1762) ~[amu.class:?]
    at net.minecraft.world.WorldServer.updateEntities(WorldServer.java:613) ~[oo.class:?]
    at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:767) ~[MinecraftServer.class:?]
    ... 4 more
    ```
    
    Reproduction steps
   1. Spawn a snow queen
   2. Get a flame bow
   3. Launch an arrow at the snow queen using a flame bow
   
  Congratulations, you are no longer able to boot up your world. Or server for that particular user.
  Behavior appears to be caused by the block fire config.

**Comment by Knifepony** [comment]
This is on version 1.4.1, an outdated version. Seems to be fixed on 1.5.0
Never mind, I'll just bug Pansmith to update UT in E2Eu.

**Comment by ACGaming** [comment]
Yes, this should indeed be fixed in the latest update!
